plugins/license-fact-providers/scancode/src/main/kotlin/ScanCodeLicenseFactProvider.kt

```diff
@@ -0,0 +1,131 @@
+    factory = LicenseFactProviderFactory::class
+)
+class ScanCodeLicenseFactProvider(
+    override val descriptor: PluginDescriptor = ScanCodeLicenseFactProviderFactory.descriptor,
+    private val config: ScanCodeLicenseFactProviderConfig
+) : LicenseFactProvider {
+    /**
+     * The directory that contains the ScanCode license texts. This is located using a heuristic based on the path of
+     * the ScanCode binary.
+     */
+    private val scanCodeLicenseTextDir: File by lazy {
+        if (config.scanCodeLicenseTextDir != null) {
+            File(config.scanCodeLicenseTextDir).takeIf { it.isDirectory }.also {
+                requireNotNull(it) {
+                    "Configured ScanCode license text directory '${config.scanCodeLicenseTextDir}' does not exist or " +
+                        "is not a directory."
+                }
+
+                logger.debug { "Using configured ScanCode license text directory: ${it.absolutePath}" }
+                return@lazy it
+            }
+        }
+
+        logger.debug { "Trying to locate the ScanCode license text directory..." }
+
+        val scanCodeExeDir = Os.getPathFromEnvironment("scancode")?.realFile?.parentFile
+
+        if (scanCodeExeDir == null) {
+            logger.debug { "Could not locate the ScanCode executable directory." }
+        } else {
+            logger.debug { "Located ScanCode executable directory: ${scanCodeExeDir.absolutePath}" }
+        }
+
+        val pythonBinDir = listOf("bin", "Scripts")
+        val scanCodeBaseDir = scanCodeExeDir?.takeUnless { it.name in pythonBinDir } ?: scanCodeExeDir?.parentFile
+
+        if (scanCodeBaseDir == null) {
+            logger.debug { "Could not locate the ScanCode base directory." }
+        } else {
+            logger.debug { "Located ScanCode base directory: ${scanCodeBaseDir.absolutePath}" }
+        }
+
+        val licenseDir =
+            scanCodeBaseDir?.walkTopDown()?.find { it.isDirectory && it.endsWith("licensedcode/data/licenses") }
+
+        if (licenseDir == null) {
+            logger.debug { "Could not locate the ScanCode license text directory." }
+        } else {
+            logger.debug { "Located ScanCode license text directory: ${licenseDir.absolutePath}" }
+            return@lazy licenseDir
+        }
+
+        File(FALLBACK_DIR).takeIf { it.isDirectory }.also {
+            if (it == null) {
+                logger.debug { "Could not locate fallback directory: $FALLBACK_DIR" }
+            } else {
+                logger.debug { "Located fallback ScanCode license text directory: ${it.absolutePath}" }
+                return@lazy it
+            }
+        }
+
+        error("Could not locate the ScanCode license text directory.")
+    }
+
+    private fun getLicenseTextFile(licenseId: String): File? {
+        val filename = "${licenseId.removePrefix("LicenseRef-").lowercase()}.LICENSE"
```

Comment:
We should probably use the `LICENSE_REF_PREFIX` constant. Or didn't you want to introduce a dependency on `spdx-utils`?